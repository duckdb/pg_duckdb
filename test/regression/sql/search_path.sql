CREATE TABLE t(a INT);
INSERT INTO t SELECT g from generate_series(1,10) g;
SELECT count(*) FROM t;
SELECT count(*) FROM public.t;

-- Create schema `other`
CREATE SCHEMA other;
CREATE TABLE other.t(a INT);
INSERT INTO other.t SELECT g from generate_series(1,100) g;
SELECT count(*) FROM other.t;

-- Test fully qualified table name combinations
SELECT count(*) FROM public.t, other.t;
SELECT count(*) FROM t, other.t;
SELECT count(*) FROM t,t;

-- search_path ORDER matters.
SET search_path TO other, public;
SELECT count(*) FROM t;
SELECT count(*) FROM t, public.t;

-- No search_path
SET search_path TO '';
SELECT count(*) FROM t, other.t;
SELECT count(*) FROM public.t, other.t;

-- Cleanup
DROP TABLE other.t;
DROP SCHEMA other;
RESET search_path;

drop table t;
