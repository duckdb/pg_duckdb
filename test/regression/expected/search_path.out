CREATE TABLE t(a INT);
INSERT INTO t SELECT g from generate_series(1,10) g;
SELECT count(*) FROM t;
 count_star() 
--------------
           10
(1 row)

SELECT count(*) FROM public.t;
 count_star() 
--------------
           10
(1 row)

-- Create schema `other`
CREATE SCHEMA other;
CREATE TABLE other.t(a INT);
INSERT INTO other.t SELECT g from generate_series(1,100) g;
SELECT count(*) FROM other.t;
 count_star() 
--------------
          100
(1 row)

-- Test fully qualified table name combinations
SELECT count(*) FROM public.t, other.t;
 count_star() 
--------------
         1000
(1 row)

SELECT count(*) FROM t, other.t;
 count_star() 
--------------
         1000
(1 row)

SELECT count(*) FROM t,t;
ERROR:  table name "t" specified more than once
-- search_path ORDER matters.
SET search_path TO other, public;
SELECT count(*) FROM t;
 count_star() 
--------------
          100
(1 row)

SELECT count(*) FROM t, public.t;
 count_star() 
--------------
         1000
(1 row)

-- No search_path
SET search_path TO '';
SELECT count(*) FROM t, other.t;
ERROR:  relation "t" does not exist
LINE 1: SELECT count(*) FROM t, other.t;
                             ^
SELECT count(*) FROM public.t, other.t;
 count_star() 
--------------
         1000
(1 row)

-- Cleanup
DROP TABLE other.t;
DROP SCHEMA other;
RESET search_path;
drop table t;
